analyzer/src/main/kotlin/managers/Carthage.kt

```diff
@@ -0,0 +1,258 @@
+
+        dependencyLines.forEach { line ->
+            if (line.isBlank() || line.isComment()) return@forEach
+            packages += parseDependencyLine(line, workingDir)
+        }
+
+        return packages
+    }
+
+    private fun parseDependencyLine(line: String, workingDir: String): Package {
+        val split = line.split(" ", limit = 3)
+        val type = DependencyType.valueOf(split[0].toUpperCase())
+        val id = split[1].removeSurrounding("\"")
+        val revision = split[2].removeSurrounding("\"")
+
+        return when (type) {
+            DependencyType.GITHUB -> {
+                // ID consists of github username/project or a github enterprise URL.
+                val projectUrl = if (isGithubUserAndProject(id)) {
+                    val (username, project) = id.split("/", limit = 2)
+                    "https://github.com/$username/$project"
+                } else {
+                    id
+                }
+                createPackageFromGenericGitUrl(projectUrl, revision)
+            }
+            DependencyType.GIT -> {
+                // ID is a generic git URL (https, git or ssh) or a file path to a local repository.
+                if (isFilePath(workingDir, id)) {
+                    // TODO: Carthage supports local git repositories.
+                    //       The downloader should be extended to support `file://` downloads.
+                    createPackageFromLocalRepository(id, revision)
+                } else {
+                    createPackageFromGenericGitUrl(id, revision)
+                }
+            }
+            DependencyType.BINARY -> {
+                // ID is an URL or a path to a file that contains a Carthage binary project specification.
+                if (isFilePath(workingDir, id)) {
+                    val filePath = id.removePrefix("file://")
+                    val binarySpecFile = when {
+                        File(filePath).isAbsolute -> File(filePath)
+                        else -> File("$workingDir/$filePath")
+                    }
+                    val content = binarySpecFile.readText()
+                    val binarySpec = jsonMapper.readValue<Map<String, String>>(content)
+                    createPackageFromBinarySpec(binarySpec, id, revision)
+                } else {
+                    val content = URL(id).readText()
+                    val binarySpec = jsonMapper.readValue<Map<String, String>>(content)
+                    createPackageFromBinarySpec(binarySpec, id, revision)
+                }
+            }
+        }
+    }
+
+    private fun createPackageFromGenericGitUrl(projectUrl: String, revision: String): Package {
+        val vcsInfoFromUrl = VcsHost.toVcsInfo(projectUrl)
+        val vcsInfo = VcsInfo(
+            type = vcsInfoFromUrl.type,
+            url = vcsInfoFromUrl.url,
+            revision = revision
+        )
+        val vcsHost = VcsHost.toVcsHost(URI(vcsInfo.url))
```

Comment:
Maybe using `String` as a param for `toVcsHost`really makes sense as many callers seem to just construct this URI.
I've commented this on the commit which creates that function.